### ISSUE TITLE
 Subscription for pagination

### ISSUE THREAD

**Issue by khriztianmoreno** [aslagle/reactive-table#351]
Currently my collection has about 4 million of documents, I would like to subscribe to the collection but only for every pagination, I mean an initial subscription of 10 documents, then you go to the next page, subscribe to documents 11 to 20.
How can I accomplish this?

**Comment by aslagle** [comment]
You can use [ReactiveTable.publish](https://github.com/aslagle/reactive-table#server-side-pagination-and-filtering-beta) on the server to set up a paginated publication that the table can use.

**Comment by ababba15** [comment]
@aslagle can you please provide more info about how to setup  a paginated publications? In your example I see only publications with some exact fields.

**Comment by aslagle** [comment]
You have to pass in a name as the first argument and use it as the collection in the client side table settings.

So `ReactiveTable.publish('places', Places, {archive: 0})`, and then the `collection` in the table settings would be `"places"`. You do have to specify a list of fields on the client when you use this - it can't automatically get them from the collection.

**Comment by ababba15** [comment]
@aslagle have done like you said and it works, thank you for reply without delay. 

But now I get another issue with that subscription — when I load exactly that page with table (`Places` table), it works fine. But if I go to another route and come back to `Places` (or just come to route with `Places` from any other route), the table doesn't load. 

Here is some DDP subscriptions log, if I can say so.

![scr 2016-06-08 at 04 45 43](https://cloud.githubusercontent.com/assets/13153161/15880370/407804f8-2d35-11e6-8fe2-3552cf6ec56c.png)

I would be happy give you more info, if you need. Please, can you suggest what can cause this behavior.

**Comment by aslagle** [comment]
It looks like the table is loading when you go back to the places route, but then it's unsubscribing for some reason. It's possible there's a bug in the reactive-table subscription code that's making it unsubscribe, but I think it's more likely the template is getting destroyed. Maybe the blaze inspector would show that. Which router are you using?

For checking if the subscriptions are ready, you can pass in a `ready` ReactiveVar in the table settings that's set to false, and it will change to true when the initial subscription is ready.

**Comment by ababba15** [comment]
No, Template is not destroing. I have removed `{{#if ready}}` inside `reactive-table.html` and table is rendering, but without any data.

**Comment by aslagle** [comment]
In that case, I would guess something's going wrong in the onReady function at line 44 of reactive-table.js, but I'm not sure what it could be.

**Comment by ababba15** [comment]
By the way, I tried to comment out all the code with `stop()`, but behavior didn't change at all. I thought that it will keep data in minimongo, but it appears and in 100ms dissapears from minimongo.

**Comment by ababba15** [comment]
Would you like to take a look what is going on? I can share my localhost via `ngrok`

**Comment by aslagle** [comment]
Sure, I can take a look.

**Comment by ababba15** [comment]
One moment

**Comment by ababba15** [comment]
I didn't minify all the code, maybe it will be useful for you.
[Here we go](http://9b3894a8.ngrok.io/admin/allLocations)

**Comment by ababba15** [comment]
Just refresh page to let the data appear again

**Comment by aslagle** [comment]
Could you try changing this line https://github.com/aslagle/reactive-table/blob/master/lib/reactive_table.js#L70 to `{onReady: onReady, onStop: onError}`. 

I see the meteor documentation has onStop.. I wonder if that changed or I just had it wrong the whole time.

**Comment by ababba15** [comment]
Done

**Comment by aslagle** [comment]
Now your app is crashing :(

**Comment by ababba15** [comment]
Yep sorry, my fall, just fixed

**Comment by aslagle** [comment]
Okay, I've set break points every place reactive-table stops a subscription and it's not any of them. There must be something else that's stopping the subscription.

**Comment by ababba15** [comment]
In my code, you mean?

**Comment by aslagle** [comment]
Or in another package

**Comment by ababba15** [comment]
hm okay, so there is no problem with your package, as I understand. So thank you for your time. Anyway it is a great package, I choose it from 3 another as the best worker with big collections.

I will keep searching, and let you know about results.. Thank you, Amy and have a nice day 👍

**Comment by aslagle** [comment]
As far as I can tell this is getting triggered by flow router calling Tracker.invalidate, but I'm not familiar enough with flow router to know what's going on. Hope you can figure it out :)

**Comment by ababba15** [comment]
Yes, Amy, you are right. It was FlowRouter. Not sure, what exactly cause this behavior, but changing `FlowRouter.go()` to `BlazeLayout.render()` fix it.

**Comment by ababba15** [comment]
Also one more little question — empty collections in minimongo don't go away while I am walking across the app. Can it cause any perfomance issues if their amount will grow (without refreshing all app)?

![image](https://cloud.githubusercontent.com/assets/13153161/15917850/060e3d44-2e09-11e6-9881-09162b98820e.png)

**Comment by aslagle** [comment]
That's a good question, I don't know what MiniMongo does with the empty collections internally so I'm not sure if it could be a problem. I don't think there's a way to remove them, but if there is, let me know.

**Comment by ababba15** [comment]
Found how to fix my problem by calling `FlowRouter.reload();` after `FlowRouter.go();`, works fine.
